src/swupd_activator.cpp

```diff
@@ -0,0 +1,486 @@
+                // }
+                if (prop == dbus::software::properties::activation)
+                {
+                    status = std::get<std::string>(value);
+                    activation =
+                        Activation::convertActivationsFromString(status);
+                    const size_t pos = status.find_last_of(".");
+                    if (pos != std::string::npos)
+                    {
+                        status.erase(0, pos + 1);
+                    }
+                }
+                if (prop == dbus::software::properties::reqActivation)
+                {
+                    reqActivation =
+                        Activation::convertRequestedActivationsFromString(
+                            std::get<std::string>(value));
+                }
+                if (prop == dbus::association::assoc)
+                {
+                    std::vector<Association> assoc;
+                    assoc = std::get<std::vector<Association>>(value);
+                    for (const auto& [fwd, rev, objPath] : assoc)
+                    {
+                        if (fwd == "inventory")
+                        {
+                            inventoryItem = fs::path(objPath).filename();
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+        catch (const sdbusplus::exception::exception& ex)
+        {
+            fprintf(stderr, "Error while reading version information: %s\n",
+                    ex.what());
+            return std::map<std::string, Version>();
+        }
+
+        if ((inventoryItem.empty()) ||
+            ((!target.empty()) && (inventoryItem != target)))
+        {
+            continue;
+        }
+        if (targetId.empty())
+        {
+            targetId = inventoryItem;
+        }
+        else
+        {
+            targetId = inventoryItem + "/" + targetId;
+        }
+        versions[verId].version = ver;
+        versions[verId].status = status;
+        versions[verId].targets.emplace(inventoryItem);
+        versions[verId].objects.emplace_back(targetId, owner, path, ver,
+                                             activation, reqActivation);
+    }
+    return versions;
+}
+
+bool printVersionsList(const std::string& versionId, const std::string& target,
+                       bool active = false)
+{
+    auto versions = getVersions(versionId, target, active);
+    fprintf(stdout, "Software package ID\t\tSoftware "
+                    "version\t\t\tStatus\t\tTarget inventory items\n");
+    for (const auto& [verId, ver] : versions)
+    {
+        std::string inventory;
+        for (std::string item : ver.targets)
+        {
+            if (!inventory.empty())
+            {
+                inventory += ", ";
+            }
+            inventory += item;
+        }
+        if (ver.version.length() <= 35)
+        {
+            fprintf(stdout, "%-24s\t%-35s\t%-10s\t%s\n", verId.c_str(),
```

Comment:
I still dislike that this line doesn't fit into 80 characters.
`24 + 8(tab) + 35 + 5(tab) + 10 = 82` plus `Target inventory items`, that's  `6(tab) + unlimited string`.
It will look weird via serial console.